apps/coin/contracts/Coin.sol

```diff
@@ -0,0 +1,50 @@
+pragma solidity ^0.4.24;
+
+import "@aragon/os/contracts/apps/AragonApp.sol";
+import "@aragon/os/contracts/lib/math/SafeMath.sol";
+
+import "./StandardCoin.sol";
+
+contract Coin is StandardCoin, AragonApp {
+    
+    using SafeMath for uint256;
+
+    bytes32 public constant ISSUE_ROLE = keccak256("ISSUE_ROLE");
+    bytes32 public constant MINT_ROLE = keccak256("MINT_ROLE");
+    //bytes32 public constant TRANSFER_ROLE = keccak256("TRANSFER_ROLE");
+    //bytes32 public constant ASSIGN_ROLE = keccak256("ASSIGN_ROLE");
+
+    string private _name;
+    string private _symbol;
+    uint8 private _decimals;
+
+    function initialize() public onlyInit {
+
+        initialized();
+    }
+
+    function name() public view isInitialized returns(string) {
+        return _name;
+    }
+
+    function symbol() public view isInitialized returns(string) {
+        return _symbol;
+    }
+
+    function decimal() public view isInitialized returns(uint8) {
+        return _decimals;
+    }
+
+    function issueToken(string memory name, string memory symbol, uint8 decimals) public isInitialized auth(ISSUE_ROLE) {
+        _name = name;
+        _symbol = symbol;
+        _decimals = decimals;
```

Comment:
we can...
What I thought about is when a member create a space, we initiate a DAO with all the apps(space, coin...ect), even if a member just create a space without w issuing a coin(without the need of the coin app), we initiate that app also, so in this way if we want to put that in the initialize(), we will have to pass those arguments to the newInstance() in the dao kit, and that's what I didn't like...
So in this case if we will do that way with another function to issue the token, I will have like to add a mechanism to insure that this function get called only one time(with a bool var maybe...)

Reply:
ok, I see. that's a valid point. 
even though I'd say we should try to prevent exceptions and options and ideally always apply the same, most simple standard process.